lib/streamlit/connections/__init__.py

```diff
@@ -14,6 +14,9 @@
 
 # Explicitly re-export public symbols.
 from streamlit.connections.base_connection import BaseConnection as BaseConnection
+from streamlit.connections.snowflake_connection import (
+    SnowflakeConnection as SnowflakeConnection,
```

Comment:
Just curious why you name it the same thing this way?

Reply:
...and below.  I just looked at the original source and this seems to be a common pattern.  Is this done for linting purposes?  Because it really isn't necessary for re-export, and definitely seems odd.  If you're doing it to avoid some linter warnings, it might be better to just explicitly ignore the linter warnings through file pragmas, which at least would be self explanatory.  E.g. We ignore [F401](https://github.com/snowflakedb/snowapi-python/blob/main/ruff.toml#L27) in snowapi-python.

Reply:
I believe this is needed by `mypy` so that it knows which parts of a module should be considered part of the public interface.

I'll punt on making changes to this in this PR since there are other places in the codebase that would need to be changed if we are able to get rid of it (although I think we'll need to keep it around for the typechecker), but I'll either see whether we can get rid of it or will improve the comments around these reexports in a followup

Reply:
A `__all__` should do the trick for mypy IIRC.  What doesn't work yet in mypy is [computed dunder-all](https://github.com/python/mypy/issues/15300).

Reply:
ah, I was unaware -- thanks! I'll address this throughout the codebase in a future PR to `develop`